drivers/docker/reconciler.go

```diff
@@ -0,0 +1,174 @@
+package docker
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	docker "github.com/fsouza/go-dockerclient"
+)
+
+func (d *Driver) removeDanglingContainersGoroutine() {
+	if !d.config.GC.DanglingContainers.Enabled {
+		d.logger.Debug("skipping dangling containers handling; is disabled")
+		return
+	}
+
+	period := d.config.GC.DanglingContainers.period
+
+	succeeded := true
+
+	// ensure that we wait for at least a period or creation timeout
+	// for first container GC iteration
+	// The initial period is a grace period for restore allocation
+	// before a driver may kill containers launched by an earlier nomad
+	// process.
+	initialDelay := period
+	if d.config.GC.DanglingContainers.creationTimeout > initialDelay {
+		initialDelay = d.config.GC.DanglingContainers.creationTimeout
+	}
+
+	timer := time.NewTimer(initialDelay)
+	for {
+		select {
+		case <-timer.C:
+			if d.previouslyDetected() && d.fingerprintSuccessful() {
```

Comment:
Is there a reason to do this inside the loop rather than once at the start?

Reply:
Yes.  As fingerprint continues, detection and healthy may change and I only want to kill containers if docker is detected and healthy at the time.  I didn't want to have fingerprinter launch goroutines and deal with shutdown, so thought that having a long running goroutine that periodically check health is a much simpler solution.

Reply:
Yeah that totally makes sense.